craft_parts/executor/part_handler.py

```diff
@@ -196,11 +202,14 @@ def _run_stage(self, step_info: StepInfo) -> StepState:
             work_dir=self._part.stage_dir,
         )
 
+        overlay_hash = self._compute_layer_hash(for_all_parts=True)
```

Comment:
Similar to above, the reason is probably different

Reply:
Also added an explanatory comment. The reason is similar except that in build we verify the overlay hash to invalidate parts with overlay visibility (so these parts are rebuilt) and in stage we verify it to invalidate parts declaring overlay parameters (so parts are restaged).